conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1266,24 +1270,34 @@ def build_requirements(self):
         find_program(FOOBIN_EXECUTABLE foobin)
         message("foobin executable: ${FOOBIN_EXECUTABLE}")
         message("foobar include dir: ${foobar_INCLUDE_DIR}")
+        if(NOT FOOBIN_EXECUTABLE)
+          message(FATAL_ERROR "FOOBIN executable not found")
+        endif()
     """)
 
     client.save({
         "conanfile_consumer.py": conanfile_consumer,
         "CMakeLists.txt": cmakelists_consumer,
         "host_profile": host_profile,
         "build_profile": build_profile}, clean_first=True)
-    client.run("install conanfile_consumer.py pkg/0.1@ -g CMakeToolchain -g CMakeDeps -pr:b build_profile -pr:h host_profile")
-
-    build_context_package_id = "581814504b2e960b35df487e5bdb32b1ecf02253"
-    host_context_package_id = "bf544cd3bc20b82121fd76b82eacbb36d75fa167"
+    
+    if single_profile:
+        profiles = "-pr host_profile"
+    else:
+        profiles = "-pr:b build_profile -pr:h host_profile"
+    
+    client.run(f"install conanfile_consumer.py pkg/0.1@ -g CMakeToolchain -g CMakeDeps {profiles}")
 
     with client.chdir("build"):
         client.run_command("cmake .. -DCMAKE_TOOLCHAIN_FILE=Release/generators/conan_toolchain.cmake -DCMAKE_BUILD_TYPE=Release")
         # Verify binary executable is found from build context package,
         # and library comes from host context package
-        assert f"package/{build_context_package_id}/bin/foobin" in client.out
-        assert f"package/{host_context_package_id}/include" in client.out
+
+        if not single_profile:
+            build_context_package_id = "581814504b2e960b35df487e5bdb32b1ecf02253"
+            host_context_package_id = "bf544cd3bc20b82121fd76b82eacbb36d75fa167"
+            assert f"package/{build_context_package_id}/bin/foobin" in client.out
+            assert f"package/{host_context_package_id}/include" in client.out
```

Comment:
Despite being using these lines:

```cmake
        if(NOT FOOBIN_EXECUTABLE)
          message(FATAL_ERROR "FOOBIN executable not found")
        endif()
```
Should we add the asserts if `single_profile`? Something like:

```python
      else:
            host_context_package_id = "bf544cd3bc20b82121fd76b82eacbb36d75fa167"
            assert f"package/{host_context_package_id}/bin/foobin" in client.out
            assert f"package/{host_context_package_id}/include" in client.out
```
Asking because I don't know if I'm missing something... 😓